src/items.rs

```diff
@@ -112,6 +139,40 @@ pub enum Formula {
     Eq(FormulaList),
 }
 
+impl Display for Formula {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Prefix
+        match self {
+            Formula::Lit(literal) => write!(f, "{}", literal)?,
+            Formula::Paren(formula) => write!(f, "({})", formula)?,
+            Formula::Neg(formula) => write!(f, "-{}", formula)?,
+            Formula::And(_) => write!(f, "*(")?,
+            Formula::Or(_) => write!(f, "+(")?,
+            Formula::Xor(_) => write!(f, "xor(")?,
+            Formula::Eq(_) => write!(f, "=(")?,
+        };
+
+        // Suffix
+        match self {
+            Formula::Lit(_) | Formula::Paren(_) | Formula::Neg(_) => {}
+            Formula::And(formula_list)
+            | Formula::Or(formula_list)
+            | Formula::Xor(formula_list)
+            | Formula::Eq(formula_list) => {
+                let fl = formula_list;
+                for formula in fl[0..fl.len() - 1].iter() {
+                    write!(f, "{} ", formula)?;
+                }
+                if !fl.is_empty() {
+                    write!(f, "{}", fl.last().unwrap())?;
+                }
```

Comment:
```suggestion
                if let Some((head, last)) = fl.split_last() {
                    for formula in rest {
                        write!(f, "{} ", formula)?;
                    }
                    write!(f, "{}", last)?;
                }
```

Reply:
not sure if that compiles but that's how I imagined it with my earlier comment. ;)

Reply:
Made the change, Robin.

Reply:
can't see it yet